pkg/utils/gardener/shootstate/shootstate.go

```diff
@@ -125,28 +126,32 @@ func computeGardenerData(
 	if err != nil {
 		return nil, err
 	}
+	// Deactivate this for now as there are now MachineDeployments in self-hosted shoots for unmanaged infrastructure case.
+	if !gardenletutils.IsResponsibleForSelfHostedShoot() {
```

Comment:
Nit:

```suggestion
	// Deactivate this for now as there are no MachineDeployments in self-hosted shoots for unmanaged infrastructure case.
	if !gardenletutils.IsResponsibleForSelfHostedShoot() {
```

However, what about the case of managed infrastructure. My understanding is that we exit here for all self-hosted clusters, but we only want to exit for self-hosted clusters with unmanaged infrastructure. This does not look right.

Reply:
Maybe we should exit the function early for certain self-hosted clusters instead of doing the default case in an `if` block.

Reply:
This is actually a mistake from my side 😅
This if branch should be removed again.

It should try to find out if the MCM resource groups, like `MachineDeployments` are there or not. If they are not present (like in the unmanaged infrastructure case), it should just skip adding them to the `ShootState` (hence the `machineState != nil`) check.
For the managed infrastructure case, they should be present IIUC, so they should be also added there.